File: python_part/Leetcode/Array/easy/1. Two sum/Solution1.py

#最垃圾的方法  需要两个for循环逐个相加后与target对比， 因此循环多少次，就相加多少次。
class Solution:
    def twoSum(self, nums, target):
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i]+nums[j] == target:
                    return [i, j]

File: Cpp_part/Leetcode/Leetcode/array/167. 两数之和 II - 输入有序数组/Solution1.cpp
#include <iostream>
#include <vector>
using namespace std;
//超时法
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> res;
        for(int i = 0; i < numbers.size(); i ++){
            res.push_back(target - numbers[i]);
        }
        for(int i = 0; i < numbers.size(); i++){
            for (int j = i+1; j < numbers.size(); j++){
                if (res[i] == numbers[j])
                    return vector<int> {i+1, j+1};
            }
        }
        return vector<int>{-1,-1};
    }
};

//int main(){
//    Solution* s = new Solution;
//    vector<int> input = {2, 7, 15, 11};
//    vector<int> res = s -> twoSum(input, 9);
//}
